engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -298,6 +298,7 @@ object IR {
     * @param passData the pass metadata associated with this node
     * @param diagnostics compiler diagnostics for this node
     */
+  @SerialVersionUID(3692L) // prevents reading broken caches, see PR-3692 for details
```

Comment:
When CI fails due to errors caused by reading previous `IR` caches, change the `3692L` value to a different unique value. Number of your PR should be unique enough.

Reply:
In my understanding the caching logic was:
- try to deserialize the IR
- if it fails, compile and re-populate the cache

You mean that here are cases when this approach doesn't work?

Reply:
There are many examples where it didn't work. Whole week I was fighting with failing CI: https://github.com/enso-org/enso/runs/8190285398?check_suite_focus=true
```
INFO ide_ci::program::command: sbtℹ️ [error] Test org.enso.interpreter.test.VectorTest.passingListOrArrayToEnsoAsArray failed: org.graalvm.polyglot.PolyglotException: org.enso.compiler.exception.CompilerError: Compiler Internal Error: Constructor for Cons(Vector_Data,1,false) is null, took 5.316 sec
 INFO ide_ci::program::command: sbtℹ️ [error]     at org.enso.compiler.codegen.IrToTruffle$ExpressionProcessor.processName(IrToTruffle.scala:1190)
 INFO ide_ci::program::command: sbtℹ️ [error]     at org.enso.compiler.codegen.IrToTruffle$ExpressionProcessor.run(IrToTruffle.scala:722)
 INFO ide_ci::program::command: sbtℹ️ [error]     at org.enso.compiler.codegen.IrToTruffle$ExpressionProcessor.run(IrToTruffle.scala:715)
```
we need a way to tell that the caches are out of date, especially when they are stored in a shared location.